src/System.Text.JsonLab/System/Text/Json/Utf8JsonReader_MultiSegment.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Buffers.Reader;
+
+namespace System.Text.JsonLab
+{
+#if UseInstrumented
+    internal ref partial struct Utf8JsonReaderInstrumented
+#else
```

Comment:
BTW, this is not specific to this PR (and it's a nit), but I would keep the type names the same and just ifdef the namespace name (and then fully qualify when you want to use the instrumented one. I think it will be easier to read the ctor code if there are no ifdefs just to rename the type.

Reply:
Fixed.